backend/src/api/database-migration.ts

```diff
@@ -0,0 +1,176 @@
+import config from '../config';
+import { DB } from '../database';
+import logger from '../logger';
+
+class DatabaseMigration {
+  private static currentVersion = 1;
+  private queryTimeout = 120000;
+
+  constructor() { }
+
+  public async $initializeOrMigrateDatabase(): Promise<void> {
+    if (!await this.$checkIfTableExists('statistics')) {
+      const connection = await DB.pool.getConnection();
+      for (const query of this.getInitializeTableQueries()) {
+        await connection.query<any>({ sql: query, timeout: this.queryTimeout });
```

Comment:
@softsimon shouldn't this be in a try-catch? what happens if one of the queries fail? we'll probably want to rollback and/or exit early.